sources/refund.move

```diff
@@ -131,100 +208,171 @@ module refund::refund {
-        pool.accounting.current_liability = pool.accounting.current_liability - refund_amount;
+        let total_refunded = total_refunded_mut(&mut pool.accounting);
+        *total_refunded = *total_refunded + refund_amount;
 
-        let funds = balance::split(&mut pool.funds, refund_amount);
+        let funds = balance::split(funds_mut(&mut pool.base_pool), refund_amount);
 
-        transfer::public_transfer(coin::from_balance(funds, ctx), sender);
+        transfer::public_transfer(coin::from_balance(funds, ctx), receiver);
     }
-    
-    /// Allows an affected address to claim a boosted refund to a new address.
-    /// The refund amount is increased by a boost calculated as half of the
-    /// original refund amount. The resulting total refund corresponds to 1.5x
-    /// the actual amount lost in the loss envent.
-    /// 
-    /// This endpoint is reserved to Aldrin, i.e. the owner of the `Publisher` object.
-    /// The refund boost is limited to user who refund via the Rinbot account.
-    /// When a user claims the refund with 1.5x boost, Aldrin will call
-    /// `claim_refund_boosted` with the `affected_address` being the user address
-    /// original affected by the loss event, and `new_address` being the associated
-    /// Rinbot account address.
-    /// 
-    /// #### Panics
-    /// - If the publisher is not valid.
-    /// - If the affected address does not exist in the unclaimed refunds list.
-    public entry fun claim_refund_boosted(
-        pub: &Publisher,
+
+    // === Phase 4: Reclaim Fund ===
+
+    public entry fun reclaim_fund(
         pool: &mut RefundPool,
-        affected_address: address,
-        new_address: address,
         ctx: &mut TxContext,
     ) {
-        assert!(package::from_module<REFUND>(pub), EInvalidPublisher);
-        assert!(table::contains(&pool.unclaimed, affected_address), EInvalidAddress);
+        assert_reclaim_phase(pool);
+
+        let funders = funders_mut(&mut pool.base_pool);
+        assert!(table::contains(funders, sender(ctx)), 0); // TODO: err code
 
-        let refund_amount = table::remove(&mut pool.unclaimed, affected_address);
-        let boost = div(refund_amount, 2);
-        let boosted_refund_amount = refund_amount + boost;
+        let total_raised = total_raised(&pool.accounting);
+        let funding_amount = table::remove(funders, sender(ctx));
 
-        pool.accounting.total_refunded = pool.accounting.total_refunded + refund_amount;
-        pool.accounting.total_boosted = pool.accounting.total_boosted + boost;
-        pool.accounting.current_liability = pool.accounting.current_liability - refund_amount;
+        let is_last = table::is_empty(funders);
+        let funds = funds_mut(&mut pool.base_pool);
+        
+        let reclaim_amount = if (is_last) {
+            balance::value(funds)
+        } else {
+            let leftovers = balance::value(funds);
```

Comment:
I think this is not what you want. Assume
- Alice, Bob, and Claire both contributed 100
- there is 10 in `leftovers` at the beginning of the reclaim phase
- Alice withdraws first, Bob withdraws second

, then I think Alice will end up getting a larger share than Bob. Instead, I think what makes sense is to record the value of `balance::value(funds)` at the beginning of the reclaim phase, and always use that (static) value here, rather than recomputing on each call to `reclaim_fund`

Reply:
I see what you're saying. I will change this.